src/devices/video/huc6270.h

```diff
@@ -4,6 +4,271 @@
+	*1 LSB of pattern select bit is select (SP0, SP1) or (SP2, SP3).
+	*2 SPO-SP3 are fetched during two consecutive character cycles.
+
+	VM assignment
+	+-----------+--------------+-----------------------------------------------+
+	|  VM bit   |              |  Assignment for one character cycle (8 dot)   |
+	+-----+-----+ Access width +-----+-----+-----+-----+-----+-----+-----+-----+
+	|  1  |  0  |              |  0  |  1  |  2  |  3  |  4  |  5  |  6  |  7  |
+	+-----+-----+--------------+-----+-----+-----+-----+-----+-----+-----+-----+
+	|  0  |  0  |      1       | CPU | BAT | CPU |  -  | CPU | CG0 | CPU | CG1 |
+	+-----+-----+--------------+-----+-----+-----+-----+-----+-----+-----+-----+
+	|  0  |  1  |      2       |    BAT    |    CPU    |    CG0    |    CG1    |
+	+-----+-----+--------------+-----+-----+-----+-----+-----+-----+-----+-----+
+	|  1  |  0  |      2       |    BAT    |    CPU    |    CG0    |    CG1    |
+	+-----+-----+--------------+-----------+-----------+-----------+-----------+
+	|  1  |  1  |      4       |          BAT          |       CG0 or CG1      |
+	+-----+-----+--------------+-----------------------+-----------------------+
+
+	BAT data (single word per tile)
+	Bit               Description
+	fedcba98 76543210
+	xxxx---- -------- CG color
+	----xxxx xxxxxxxx Character code select (16 word each)
+
+	SAT data (4 word per each sprites)
+	Offset Bit               Description
+	       fedcba98 76543210
+	00     ------xx xxxxxxxx Y coordinate - 64
+	01     ------xx xxxxxxxx X coordinate - 32
+	02     -----xxx xxxxxxxx PC (Pattern code, 64 word each)
+	       -------- -------x SG0/SG1, SG2/SG3 select, for 4 color mode
+	03     x------- -------- Flip Y
+	       --xx---- -------- CGY (Sprite height)
+		   --00---- -------- 1 tile (16 pixel)
+		   --01---- -------- 2 tiles (32 pixel)
+		   --10---- -------- Not used
+		   --11---- -------- 4 tiles (64 pixel)
+		   ----x--- -------- Flip X
+		   -------x -------- CGX (Sprite width)
+		   -------0 -------- 1 tile (16 pixel)
+		   -------1 -------- 2 tiles (32 pixel)
+		   -------- x------- SPBG (Sprite VS Background priority)
+		   -------- 0------- Behind background
+		   -------- 1------- Above background
+		   -------- ----xxxx SPRITE COLOR
+
+	VD0-7, SPBG output
+	Background:
+	Bit       Description
+	8 7654 3210
+	0 ---- ---- 0 for background
+	- xxxx ---- CG Color (when VD0-VD3 != 0)
+	- ---- x--- CG3 (or 0 for 4 clocks, CG0 selected)
+	- ---- -x-- CG2 (or 0 for 4 clocks, CG0 selected)
+	- ---- --x- CG1 (or 0 for 4 clocks, CG1 selected)
+	- ---- ---x CG0 (or 0 for 4 clocks, CG1 selected)
+	- ---- 0000 Transparency, bit 7-4 is forced to 0
+
+	Sprite:
+	Bit       Description
+	8 7654 3210
+	1 ---- ---- 1 for sprite
+	- xxxx ---- Sprite Color (when VD0-VD3 != 0)
+	- ---- x--- SG3
+	- ---- -x-- SG2
+	- ---- --x- SG1
+	- ---- ---x SG0
+	- ---- 0000 Transparency, bit 7-4 is forced to 0
+
+	Border:
+	1 0000 0000 Border color, during blanking period
+
```

Comment:
Totally not a fan of having register implementation details in header for visibility reasons, every description should really go near the relevant register fn or at top of .cpp header instead.